From 5b8e38bf81f50472f0090bed413255093f3e4270 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <vonreth@kde.org>
Date: Tue, 19 Dec 2017 17:54:05 +0100
Subject: [PATCH] Don't cause circular linking on Windows

Summary:
When building plugins, don't arbitrarily rename the output files (as this will occasionally result in circular dependencies).

[[ https://git.reviewboard.kde.org/r/118084/ | In this review from three and a half years ago ]], the Sonnet plugins were moved into a subdirectory, which was great. They were, however, also renamed at the same time, which resulted in some (and specifically the aspell plugin) being named the same as the library they should be linked against. End result: Spell checking has not worked on Windows for three and a half years.

Test Plan:
With these properties set, the plugins fail to load on Windows due to circular dependencies

Without them (that is, with this patch), the plugins load

Reviewers: #frameworks

Subscribers: vonreth, cgiboudeaux, alexeymin, apol, #frameworks

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D9018
---
 src/plugins/aspell/CMakeLists.txt         | 1 -
 src/plugins/enchant/CMakeLists.txt        | 1 -
 src/plugins/hspell/CMakeLists.txt         | 1 -
 src/plugins/hunspell/CMakeLists.txt       | 1 -
 src/plugins/nsspellchecker/CMakeLists.txt | 1 -
 src/plugins/voikko/CMakeLists.txt         | 1 -
 6 files changed, 6 deletions(-)

diff --git a/src/plugins/aspell/CMakeLists.txt b/src/plugins/aspell/CMakeLists.txt
index 81b423d..3b6ad96 100644
--- a/src/plugins/aspell/CMakeLists.txt
+++ b/src/plugins/aspell/CMakeLists.txt
@@ -14,5 +14,4 @@ add_library(sonnet_aspell MODULE ${sonnet_aspell_PART_SRCS})

 target_link_libraries(sonnet_aspell PRIVATE KF5::SonnetCore ${ASPELL_LIBRARIES})

-set_target_properties(sonnet_aspell PROPERTIES OUTPUT_NAME "aspell")
 install(TARGETS sonnet_aspell  DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/sonnet/)
diff --git a/src/plugins/enchant/CMakeLists.txt b/src/plugins/enchant/CMakeLists.txt
index 3283d24..72d6d65 100644
--- a/src/plugins/enchant/CMakeLists.txt
+++ b/src/plugins/enchant/CMakeLists.txt
@@ -10,6 +10,5 @@ add_library(kspell_enchant MODULE ${kspell_enchant_PART_SRCS})

 target_link_libraries(kspell_enchant PRIVATE KF5::SonnetCore ${ENCHANT_LIBRARIES})

-set_target_properties(kspell_enchant PROPERTIES OUTPUT_NAME "enchant")
 install(TARGETS kspell_enchant DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/sonnet/)

diff --git a/src/plugins/hspell/CMakeLists.txt b/src/plugins/hspell/CMakeLists.txt
index 4bf7540..c2dc2d4 100644
--- a/src/plugins/hspell/CMakeLists.txt
+++ b/src/plugins/hspell/CMakeLists.txt
@@ -18,6 +18,5 @@ add_library(sonnet_hspell MODULE ${sonnet_hspell_PART_SRCS})

 target_link_libraries(sonnet_hspell PRIVATE KF5::SonnetCore ${HSPELL_LIBRARIES} ${ZLIB_LIBRARY})

-set_target_properties(sonnet_hspell PROPERTIES OUTPUT_NAME "hspell")
 install(TARGETS sonnet_hspell  DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/sonnet/)

diff --git a/src/plugins/hunspell/CMakeLists.txt b/src/plugins/hunspell/CMakeLists.txt
index 6a6b1e1..ae64d53 100644
--- a/src/plugins/hunspell/CMakeLists.txt
+++ b/src/plugins/hunspell/CMakeLists.txt
@@ -18,7 +18,6 @@ add_library(sonnet_hunspell MODULE ${sonnet_hunspell_PART_SRCS})

 target_link_libraries(sonnet_hunspell PRIVATE KF5::SonnetCore ${HUNSPELL_LIBRARIES})

-set_target_properties(sonnet_hunspell PROPERTIES OUTPUT_NAME "hunspell")
 install(TARGETS sonnet_hunspell  DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/sonnet/)


diff --git a/src/plugins/nsspellchecker/CMakeLists.txt b/src/plugins/nsspellchecker/CMakeLists.txt
index f8c99d1..5e80a10 100644
--- a/src/plugins/nsspellchecker/CMakeLists.txt
+++ b/src/plugins/nsspellchecker/CMakeLists.txt
@@ -7,6 +7,5 @@ add_library(sonnet_nsspellchecker MODULE ${sonnet_nsspellchecker_PART_SRCS})

 target_link_libraries(sonnet_nsspellchecker PRIVATE KF5::SonnetCore "-framework Cocoa")

-set_target_properties(sonnet_nsspellchecker PROPERTIES OUTPUT_NAME "nsspellchecker")
 install(TARGETS sonnet_nsspellchecker  DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/sonnet/)

diff --git a/src/plugins/voikko/CMakeLists.txt b/src/plugins/voikko/CMakeLists.txt
index 1498824..ca22424 100644
--- a/src/plugins/voikko/CMakeLists.txt
+++ b/src/plugins/voikko/CMakeLists.txt
@@ -10,5 +10,4 @@ add_library(sonnet_voikko MODULE ${sonnet_voikko_PART_SRCS})

 target_link_libraries(sonnet_voikko PRIVATE KF5::SonnetCore ${VOIKKO_LIBRARIES})

-set_target_properties(sonnet_voikko PROPERTIES OUTPUT_NAME "voikko")
 install(TARGETS sonnet_voikko  DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/sonnet/)
--
2.15.0.windows.1

